### PR TITLE
Eval-Stig new result engine module parsing

### PR DESCRIPTION
**Pull request by Matte22**
Added changes to support parsing of newest style of Evaluate-Stig result engine data. 

This includes changes for the CKL and CKLB parser along with associated tests. 

This PR also refactors test file structuring.

### CODE REVIEW

File: ReviewParser.js [hunk 1]
```diff
@@ -338,13 +370,13 @@ export function reviewsFromCkl(
         resultEngineRoot = {
           type: 'script',
           product: 'Evaluate-STIG',
-          version: esRootComment?.global?.[0]?.version,
+          version: esRootComment?.global?.[0]?.version || esRootComment?.version,
           time: esRootComment?.global?.[0]?.time,
           checkContent: {
             location: esRootComment?.module?.[0]?.name ?? ''
```
Comment (cd-rite) [inline ReviewParser.js:376]: I think we may want to record the "module version" that is specified in the iSTIG level XML comment, as well as the module name here? (as distinct from the overall Eval-STIG "version" that performed the scan)
The spec offers "location" and "component" as properties of the ResultEngineCheckContent object.
I propose setting the value of "checkContent.component" to the specified module version.

iSTIG xml comment for reference:
`<!--<Evaluate-STIG><time>2023-12-11T12:56:14.3576272-05:00</time><module><name>Scan-GoogleChrome_Checks</name><version>1.2023.7.24</version></module></Evaluate-STIG>-->`

Comment (Matte22) [reply ReviewParser.js:376]: So for your example:
`<!--<Evaluate-STIG><time>2023-12-11T12:56:14.3576272-05:00</time><module><name>Scan-GoogleChrome_Checks</name><version>1.2023.7.24</version></module></Evaluate-STIG>-->`

desired output:
`{
        type: 'script',
        product: 'Evaluate-STIG',
        version: '1.2310.1',
        time: '2026-12-11T12:56:14.3576272-05:00',
        checkContent: {
          location: 'Test1'
          component: '1.2023.7.24'
        }
      },
}`

Is this correct?

Comment (cd-rite) [reply ReviewParser.js:376]: That said, the code does now reproduce the original output structure for single-stig ckl files, so we could create a separate issue for this.

File: ReviewParser.js [hunk 2]
```diff
@@ -214,8 +216,12 @@ export function reviewsFromCkl(
       comment
     }
 
+    // if the current checklist contrains a comment from ES, process it and add it to the review
+    const iStigCommentProcessed = iStigComment ? processIstigXmlComments(iStigComment) : null;
```
Comment (csmig) [inline ReviewParser.js:220]: Just a thought: change the implementation of  `processIstigXmlComments()` to return `null` if `iStigComment` is falsy. Then this line could be simplified to `const iStigCommentProcessed = processIstigXmlComments(iStigComment)`

File: ReviewParser.js [hunk 3]
```diff
@@ -323,6 +329,32 @@ export function reviewsFromCkl(
...
+        resultEngineIStig = {
+          time: esIStigComment?.time,
+          checkContent: {
+            location: esIStigComment?.module?.[0]?.name ?? ''
+          }
+        }
+      }
+    }
+    return resultEngineIStig || null
+  }
```
Comment (csmig) [inline ReviewParser.js:355]: Maybe return `null` if `!iStigComment`. Then any callers would not need to check if `iStigComment` exists?

File: ReviewParser.js [hunk 4]
```diff
@@ -862,38 +894,15 @@ export function reviewsFromCklb(
...
-    //     }
-    //     if (overrides.length) {
-    //       review.resultEngine.overrides = overrides
-    //     }  
-    //   }
-    // }
-    // else {
-    //   review.resultEngine = null
-    // }
+    const iStigCommentProcessed = evalStigResultEngine ? processStigEvalStigModule(evalStigResultEngine) : null;
```
Comment (csmig) [inline ReviewParser.js:897]: I have the same thought here as on line 220

File: ReviewParser.js [hunk 5]
```diff
@@ -715,8 +747,8 @@ export function reviewsFromCklb(
   if (!validationResult.valid) {
     throw (new Error(`Invalid CKLB object: ${validationResult.error}`))
   }
-
-  const resultEngineCommon = cklb.stig_manager_engine || null
+  // extract root evaluate-stig object
+  const resultEngineCommon = cklb['evaluate-stig'] ? processRootEvalStigModule(cklb['evaluate-stig']) : null
```
Comment (csmig) [inline ReviewParser.js:751]: I have the same thought here as on line 220

File: ReviewParser.js [hunk 6]
```diff
@@ -959,6 +968,33 @@ export function reviewsFromCklb(
...
+    if(module){
+      resultEngineCommon = {
+        type: 'script',
+        product: 'Evaluate-STIG',
+        version: module.version,
+      }
+    }
+    return resultEngineCommon || null
+  }
+
```
Comment (csmig) [inline ReviewParser.js:997]: Maybe implement these functions so they return `null` if the argument is falsy? See comment after line 355

**Review by csmig — APPROVED** [review]
Couple of thoughts which are not mandatory for merging but could be considered if there is time and agreement.

**Pull request merged**